source/crud/crud.rst

```diff
@@ -754,30 +754,38 @@ older sharded clusters.
 estimatedDocumentCount
 ~~~~~~~~~~~~~~~~~~~~~~
 
-On server versions greater than or equal to 4.9.0 (wire version 12 or higher),
-the estimatedDocumentCount function is implemented using the ``$collStats``
-aggregate pipeline stage with ``$group`` to gather results from multiple shards.
-As documented above, the only supported option is maxTimeMS::
-
-  pipeline = [
-    { '$collStats': { 'count': {} } },
-    { '$group': { '_id': 1, 'n': { '$sum': '$count' } } }
-  ]
-
-Similar to the count command, the estimated count of documents is returned
-in the ``n`` field. Implementations can assume that the document containing
-the single result of the aggregation pipeline is contained in the first batch of
-the server's reply to the aggregate command. It is not necessary to execute a getMore
-operation to ensure that the result is available.
-
-In the event this aggregation is run against a non-existent namespace, a NamespaceNotFound(26)
-error will be returned during execution. Drivers MUST interpret the server error code 26 as
-a ``0`` count.
-
-For server versions less than 4.9.0 (wire version 11 or under), the estimatedDocumentCount
-function is implemented using the ``count`` command with no query filter, skip,
-limit, or other options that would alter the results. Once again, the only supported
-option is maxTimeMS.
+The estimatedDocumentCount function is implemented using the ``count`` command
+with no query filter, skip, limit, or other options that would alter the
+results. The only supported options are listed in the
+``EstimatedDocumentCountOptions`` type defined above.
+
+Drivers MUST document that, due to a bug in versions 5.0.0-5.0.7 of MongoDB,
+estimatedDocumentCount was not included in v1 of the Stable API, so users of the
+Stable API are recommended to upgrade their server version to 5.0.8+ or set
+``apiStrict: false`` to avoid encountering errors when using
+estimatedDocumentCount.
+
+Drivers MUST document that estimatedDocumentCount will perform a slower but
+correct count of the documents when performed against a view. In future server
+versions, this will be updated to provide a fast but estimated count instead,
+matching the behavior of the command on regular collections.
```

Comment:
Then I'd suggest we do whatever we can to ensure driver's get notified of a future fix in the server, so we can come back and update this otherwise open-ended caveat.

Reply:
So I talked to Chris again and there is no timeline or ticket (besides definitely after 7.0), so he suggested linking to a central documentation point that will get updated as things change. The shell's documentation page has already been updated to [include this info](https://www.mongodb.com/docs/manual/reference/method/db.collection.estimatedDocumentCount/#mechanics), but I hesitate to link to there from driver docs as it can easily confuse users, especially of the Node driver. Instead, I updated the requirement to link to the [manual page for the `count` command](https://www.mongodb.com/docs/manual/reference/command/count/#behavior) itself, and if we agree with this change, I'll file a DOCS ticket to get that page updated with the same info that the shell's page has.